azavea/temperate
# Fix validation check for typeahead fields

## Overview

Using a `setTimeout` to delay the validation event on certain typeahead fields allowed the user to proceed to the next step of the wizard with an invalid field value. Removing the `setTimeout` call fixes the issue described in #495.

The `setTimeout` was introduced to fix an issue (see the commit message of 42c0576a30138503b67e0499b4a8f3285071c423). However, I believe that issue is addressed by using the typeahead name-spaced version of the blur event.

### Demo

![feb-01-2018 15-46-52](https://user-images.githubusercontent.com/1042475/35702494-4997ef90-0767-11e8-9155-b6404f522adb.gif)

## Testing Instructions

- Open the Risk wizard by editing an existing risk.
- Edit either Hazard or Community System to have text that does not match an existing value
- Without first clicking away from the typeahead, click the continue button
- Verify that a validation error appears in the field that you edited.
- Separately, attempt to recreate the error described in 42c0576a30138503b67e0499b4a8f3285071c423 and verify that you can't.

Closes #495
